src/Phinx/Db/Adapter/MysqlAdapter.php

```diff
@@ -1117,8 +1117,15 @@ protected function getColumnSqlDefinition(Column $column)
             $sqlType = $this->getSqlType($column->getType(), $column->getLimit());
             $def = strtoupper($sqlType['name']);
         }
+        
         if ($column->getPrecision() && $column->getScale()) {
```

Comment:
```suggestion
        if ($column->getPrecision()) {
```

It should be only precision check here, so it works with scale set to any number, 0, or not set.

I gave correct condition in first comment, but I made this mistake in second comment https://github.com/cakephp/phinx/issues/1481#issuecomment-454351168